test/unit/conftest.py

```diff
@@ -0,0 +1,137 @@
+
+    @staticmethod
+    def create_blockchain_wallet(*args, **kwargs):
+        return MockBlockchainTasker._blockchain_address()
+
+    @staticmethod
+    def send_eth(*args, **kwargs):
+        return MockBlockchainTasker._generic_task()
+
+    @staticmethod
+    def deploy_contract(*args, **kwargs):
+        return MockBlockchainTasker._generic_task()
+
+    @staticmethod
+    def make_token_transfer(*args, **kwargs):
+        return MockBlockchainTasker._generic_task()
+
+    @staticmethod
+    def make_approval(*args, **kwargs):
+        return MockBlockchainTasker._generic_task()
+
+    @staticmethod
+    def make_liquid_token_exchange(*args, **kwargs):
+        return MockBlockchainTasker._generic_task()
+
+    @staticmethod
+    def get_conversion_amount(from_amount, *args, **kwargs):
+        return random.random() * from_amount
+
+    @staticmethod
+    def get_token_decimals(*args, **kwargs):
+        return 18
+
+    @staticmethod
+    def get_wallet_balance(*args, **kwargs):
+        return 100
+
+    @staticmethod
+    def deploy_exchange_network(*args, **kwargs):
+        return MockBlockchainTasker._blockchain_address()
+
+    @staticmethod
+    def deploy_and_fund_reserve_token(*args, **kwargs):
+        return MockBlockchainTasker._blockchain_address()
+
+    @staticmethod
+    def deploy_smart_token(*args, **kwargs):
+        return MockBlockchainTasker._blockchain_address()
+
+    @staticmethod
+    def topup_wallet_if_required(*args, **kwargs):
+        return MockBlockchainTasker._generic_task()
+
+    @staticmethod
+    def _generic_task():
+        return random.randint(0, 10000)
+
+    @staticmethod
+    def _blockchain_address():
+        return keys.PrivateKey(os.urandom(32)).public_key.to_checksum_address()
+
+
+class MockMiscTasker(object):
+
+    @staticmethod
+    def set_ip_location(*args, **kwargs):
+        pass
+
+
+class MockEthHelper(object):
+    @staticmethod
+    def set_ip_location(*args, **kwargs):
+        pass
+
+
+@pytest.fixture(scope='module', autouse=True)
+def mock_blockchain_tasks(monkeysession):
+    from server import bt
+    mock_class(bt, MockBlockchainTasker, monkeysession)
+
+
+@pytest.fixture(scope='module', autouse=True)
+def mock_misc_tasks(monkeysession):
+    from server import mt
+    mock_class(mt, MockMiscTasker, monkeysession)
+
+
+@pytest.fixture(scope="session")
+def monkeysession(request):
+    from _pytest.monkeypatch import MonkeyPatch
+    mpatch = MonkeyPatch()
+    yield mpatch
+    mpatch.undo()
+
+
+def mock_class(orginal, substitute, monkey):
+    for func_name in dir(orginal):
+        if callable(getattr(orginal, func_name)) and not func_name.startswith("_"):
+            func = getattr(substitute, func_name, None)
+
+            if func is None:
+                raise Exception("Function {} doesn't have a mock definition in {}!".format(func_name, substitute))
+
+            monkey.setattr(orginal, func_name, func)
```

Comment:
it'd be nice to be able to move this up a directory so that the functional test can access it, but apparently you're not supposed to turn the parent directories in pytest into modules, so i don't know what to do here...

Reply:
Can you not just move this code into the `conftest.py` file located in the test directory? May need to just update the pytest fixture scope as necessary

Reply:
Also, we _can_ turn pytest directories into modules, just not recommended (and we've gotten this far without)...
http://doc.pytest.org/en/latest/goodpractices.html

Reply:
I couldn't get it to import from conftest.py. Actually fixed it just putting it in the same folder as fixtures